### DISCUSSION TITLE
Help, with new Devices, Merten MEG5126-0300 System M

### DISCUSSION THREAD

**Discussion by tech2mar** [Koenkk/zigbee2mqtt#14304] [Devices]
Hey Friends,

I got a couple of new wiser devices, but adding new ones are very hard for me, at this point i have no clue how to do that.
I can pair the device and that's it.
In the device section of zigbee2mqtt.io
there is the :
Schneider Electric MEG5113-0300/MEG5165-0000
basically the same.

This one is the : 
Merten MEG5126-0300

Schneider owns Merten.
And the Wiser Product uses Zigbee to communicate.
it would be so great too have them running.

Can someone help me with some noob proof explication?
Or maybe i can help?

Thanks

**Comment by dotnetpro100888** [comment]
Hey, were you able to pair the device? 
i have the same device and I can pair it with zigbeetomqtt, but the device not working because there isn't a converter for it. I have created one, but the device disconnect after few minutes by itself. after the device has diconnected I have to completely reset the device every time because it doesn't do anything anymore. the joining process with zigbeetomqtt seems to be getting more difficult with every try.

**Reply by tech2mar** [reply]
The device itself is working fine as a router, I have not been able to get it to work properly also, not experiencing any network related issues.

It would be great to have pro gear like that working.

**Reply by tech2mar** [reply]
https://github.com/Koenkk/zigbee2mqtt/issues/14346

**Reply by dotnetpro100888** [reply]
Based on the Informations out of the Database file you should able to create a converter for the device. Maybe i can try it and publish here.

**Reply by tech2mar** [reply]
This would be awesome.

**Reply by dotnetpro100888** [reply]
Hi, please try it with the converter below. Here you can see how to add a new converter: https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_1-pairing-the-device-with-zigbee2mqtt

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;        

const definition = {
zigbeeModel: ['2GANG/SWITCH/1'],
model: 'MEG5126-0300',
vendor: 'Schneider Electric',
description: 'Merten MEG5165 PlusLink Shutter insert with Merten Wiser System M Push Button (2fold)',
extend: extend.switch(),
exposes: [e.switch().withEndpoint('l1'), e.switch().withEndpoint('l2')],
meta: {multiEndpoint: true},
configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint1 = device.getEndpoint(1);
            await reporting.bind(endpoint1, coordinatorEndpoint, ['genOnOff']);
            await reporting.onOff(endpoint1);
            const endpoint2 = device.getEndpoint(22);
            await reporting.bind(endpoint2, coordinatorEndpoint, ['genOnOff']);
            await reporting.onOff(endpoint2);
        },
endpoint: (device) => {
            return {l1: 1, l2: 22};
        }
};

module.exports = definition;

**Reply by tech2mar** [reply]
I created the js file and pointed the add-on to it but it still didn't work, maybe I missed something.. Thanks for your help and work.

**Reply by dotnetpro100888** [reply]
Is the device now displayed as 'supported' in the device overview? Did you restart zigbeetomqtt after adding the converter?

**Reply by tech2mar** [reply]
still unsupported, maybe i made some errors.

**Reply by dotnetpro100888** [reply]
when you added the converter properly, the device should displayed as 'supported'. even if the functions of the device does not can be executed

**Reply by tech2mar** [reply]
lets see, i created a js file with the converter that you made, placed it in the zigbee2mqtt folder, pointed in the addon config that there is a external converter - reboot the addon, reboot the z2m ...and the device is not supported.

the name of the converter is also the same.. so did i miss something?

**Reply by dotnetpro100888** [reply]
from my point of view the converter did not loaded correctly. please open the configuration.yaml file in the zigbee mqtt-folder. there you must have an entry of your converter (Name of js File) like: 

![Screenshot_20221223-215352.png](https://user-images.githubusercontent.com/118766466/209404642-daf5dd50-f515-47ff-9253-291432480be0.png)

**Reply by tech2mar** [reply]
Okay, so this goes to the advenced part of the config file? Does I understand that right? And the debugger should be in debug and not in error mode?

**Reply by tech2mar** [reply]
It works, I think I figure it out, we need to merge it to the mainstream. 👌 This is great thanks

**Reply by dotnetpro100888** [reply]
Awesome! Can you Tell me what you did?

I just can't get the device to pair. the interview failed every time.

Which coordinator and which zigbeetomqtt-version do you use?

**Reply by tech2mar** [reply]
Zigbee2MQTT Version
1.28.4 commit: unknown
Coordinator-Typ
zStack3x0
Coordinator-Version
20220219
Sonoff USB Dongle P

I know that the pairing is a pain, but it is not impossible.

**Reply by dotnetpro100888** [reply]
Thank you for the Infos. I have exactly the Same  Setup in use. So this should work for me too. 

Can you Tell me how i can pair the device? How did you do it?

**Reply by tech2mar** [reply]
Normally it's just pushing 3x the 1st Button every push should be short, the problem is that you will need to reset your device.

Reseting is pushing 3x Short and the next push Holding down 10sec. The light should blink 10 times +/-.

https://download.schneider-electric.com/files?p_enDocType=User+guide&p_File_Name=MEG51xx-0300_6000_WH_DUG_DE.pdf&p_Doc_Ref=MEG51xx-0300_6000_WH_DUG_DE

**Comment by pfandfrei** [comment]
@tech2mar @dotnetpro100888 : maybe one of you has an idea what's wrong with my devices. I have 2 MEG5162 insert (2-gang relais) with a MEG5126-0300 connected module. With the information of this thread I was able to pair the module and can send and receive switching messages for both channels.
But: output 2 of the insert is always closed output 1 always open. It seems that the insert is not switched be the attached module. I'm really sure that's not a software problem here but something is wrong.
resetting the module did not help. replacing the insert also did not help

please, do you have any tips for me?

**Reply by tech2mar** [reply]
Hello, does the MEG5162 make that also with another module? like the Argus PIR?
I know for sure that the MEG5162 as 2 NO contacts, there is a possibility to create scenes with the Plus Link modules.

That sounds strange .

**Reply by pfandfrei** [reply]
@tech2mar Thank you for your answer. Currently I have no other module. I made a chat with Merten support, but he has also no idea and we agreed that this module must be defective . I will order a new one and return this one after the problem is solved

**Comment by pfandfrei** [comment]
it could have been so easy: Got a new module and it's working at first try. There are some small differences to the code posted in this thread, so I want to share my solution:
`
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
zigbeeModel: ['2GANG/SWITCH/2'],
model: 'MEG5126-0300',
vendor: 'Schneider Electric',
description: 'Merten MEG5165 PlusLink Shutter insert with Merten Wiser System M Push Button (2fold)',
extend: extend.switch(),
exposes: [e.switch().withEndpoint('l1'), e.switch().withEndpoint('l2')],
meta: {multiEndpoint: true},
configure: async (device, coordinatorEndpoint, logger) => {
const endpoint1 = device.getEndpoint(1);
await reporting.bind(endpoint1, coordinatorEndpoint, ['genOnOff']);
await reporting.onOff(endpoint1);
const endpoint2 = device.getEndpoint(2);
await reporting.bind(endpoint2, coordinatorEndpoint, ['genOnOff']);
await reporting.onOff(endpoint2);
},
endpoint: (device) => {
return {l1: 1, l2: 2};
}
};

module.exports = definition;

`

**Reply by tech2mar** [reply]
thanks, it would be great to merge this into the main ?

**Reply by Koenkk** [reply]
@pfandfrei thanks, added!

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by tech2mar** [reply]
Great, thank you all.

**Comment by pfandfrei** [comment]
@Koenkk  Hi, sorry that I haven't seen that before: Could you please change the device desciription to:

 description: 'Merten MEG5162 PlusLink relais insert with Merten Wiser System M push button (2fold)',

This was a bad copy/paste error by me

**Reply by Koenkk** [reply]
Done!

**Reply by tech2mar** [reply]
great! Thanks

**Reply by tech2mar** [reply]
Hey; Does anyone have the same issue that it stopped working after the new version?

Cheers

**Comment by Xaqumuti** [comment]
Hi,
In case it helps, you can download the Zigbee attributes specification for Schneider Electric / Merten devices from here:
https://shop.exchange.se.com/en-US/apps/103010/zigbee-interface-and-behaviour-documentation-for-schneider-electric-connected-devices
Only e-mail registration needed.
2GANG/SWITCH/2 is covered i "ZB Spec - 2g Application Module with 2ch Switch... pdf"
Cheers,
[ZB Spec - 2g Application Module with 2ch Switch - 110422.pdf](https://github.com/Koenkk/zigbee2mqtt/files/11062496/ZB.Spec.-.2g.Application.Module.with.2ch.Switch.-.110422.pdf)

**Reply by tech2mar** [reply]
@Koenkk can you help? if there is some imporvemnts that we need...

**Reply by Koenkk** [reply]
@tech2mar you should already be able to pair the device (but you have to know how to reset the device correctly as probably is documented in its manual)

**Reply by tech2mar** [reply]
Hey friends, what's strange the devices that I have paired and used with the converter I could never use without the converter after getting main support. And yes I reset the device a couple of times, it's just strange seams like a bug. I can handle it like that but it's just strange.

**Reply by linuxrrze** [reply]
I got the same device (already had the actor/relay part exchanged once, because only one of the relays seemed to be working, but the new device acted the same way).
Is there any special configuration required to make the second switch work?

**Reply by tech2mar** [reply]
With Merten Plus Link maybe... I saw something like that in the docs.

I have the same problem with one of mine, can't tell you how I did it.

I will look in the docs and report back.

**Reply by tech2mar** [reply]
So after a little research,

I found that there is a diference between Gen1 and Gen2 modules, (that's what I see)
There is a bluetooth capable one and a non bluetooth one.

For the underlying relay module the upper module (zigbee/bluetooth) seems to configure the behavior, like if you have a 2fold zb module it can act with the underlying shutter module as a shutter module.

If you pull the upper part and configured as a shutter module on a new simple 2 gang relay it will constantly blink red as it would be an error, you are forced to reconfigure (pairing mode) the upper module on the underlying module.

For me now it's a good thing to know, but on the other side in the z2mqtt herdsman side it should be considered and that's why I have some modules that are now main z2m and other modules that run on the converter because there is a diference.

If it is a 2gang underlying module or a 1gang underlying module. 1 relay or 2 relays, there is a dosen of diferent underlying modules like electronic ones dimmer, shutter, etc.. All can be used with the same basic module and can be paired, but there is definitely work to be done. I'm not a good programmer, but I can see the work.

**Reply by Nailik** [reply]
@Xaqumuti according to your file it's also possible to set SwitchIndication any idea how to integrate that into z2mqtt?

**Comment by Nailik** [comment]
I just got a Problem as well.
Pairing the MEG5126-0300 with the MEG5171-0000 (1 Gang Dimmer)
I can dim but not use the second button for anything else because it's not correctly identified.

It seems to report 3 endpoints in the log 3, 21 and 22.
I identified 3 as being the endpoint to control the dimmer.
But for 21 and 22 i don't know how to identify the supported features, for example on/off is reported as not supported.
